### ISSUE TITLE
Is DispatchEvent ts definition wrong ?

### ISSUE THREAD

**Issue by mgohin** [mpetazzoni/sse.js#65]
Hi,

In types/sse.d.ts the `DispatchEvent` is defined as this : `export type DispatchEvent = (type: string, listener: Function) => boolean;`.

In `lib/sse.js` the `dispatchEvent` function only takes the event as argument : `this.dispatchEvent = function(e) {`.

Is the TS definition wrong ?

Have a great day ;)

**Comment by mpetazzoni** [comment]
@mgohin Yes the types no longer match. If you know more about this area, would you be able to help take a look at https://github.com/mpetazzoni/sse.js/pull/59 and see if that's a change that would work? I'm not sure how to test that kind of thing myself locally.

**Comment by mgohin** [comment]
You removed the definition of `DispatchEvent` so yes it'd definitely work, but I guess you could keep definitions, just keep them updated ?

To test it, you can do like me, try to override the default function (`this.source` is the SSE instance, and my goal here is to intercept all events received) :

![image](https://github.com/mpetazzoni/sse.js/assets/3430304/b4b1c421-3a09-485b-8394-cd5fb1142513)

So here you see that if I don't use `// @ts-ignore` I got an error because my function does not match the (outdated) definition.

By the way, is there a way to intercept all events without doing like I'm doing ? I cannot subscribe to all events because I don't know all event names :)

**Comment by mpetazzoni** [comment]
Thanks @mgohin.

There is no way to (currently) get all events, see https://github.com/mpetazzoni/sse.js/issues/62#issuecomment-2086879458 for the reason why.
